Empty dataframe construction triggers multiprocessing error in pytest

### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.


### Issue description

NOTE: does not affect usage, purely the test suite!

On latest master, run pytest with warnings turned into errors:
```
pytest tests/unit/test_series.py -Werror
```

```
====================================================== short test summary info =======================================================
FAILED tests/unit/test_series.py::test_filter_ops - DeprecationWarning: passing a boolean mask to Series.__getitem__ is being depre...
FAILED tests/unit/test_series.py::test_abs - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <function Pool.__del__ ...
FAILED tests/unit/test_series.py::test_reshape - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <function Pool.__de...
FAILED tests/unit/test_series.py::test_shuffle - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <function Pool.__de...
FAILED tests/unit/test_series.py::test_duration_arithmetic - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <functi...
FAILED tests/unit/test_series.py::test_str_split - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <function Pool.__...
=================================================== 6 failed, 166 passed in 1.65s ====================================================
```

Note how all except the first test failure are due to an error in `Pool.__del__`. This error also appears in test modules other than `test_series.py`. The commonality across all the `Pool.__del__` tests is `pl.select`. In turn, that is due to it constructing a dataframe from an empty list.


### Reproducible example

Run in pytest with `-Werror`:

```python

def test_list() -> None:  # no failure
    pl.DataFrame([1,2])

def test_empty_list() -> None:  # fail, but does construct a valid dataframe
    pl.DataFrame([])

def test_dict() -> None:  # no failure
    pl.DataFrame({"a":[1]})

def test_empty_dict() -> None:  # fail, but does construct a valid dataframe
    pl.DataFrame({})
```
Example failure message:
```
E               pytest.PytestUnraisableExceptionWarning: Exception ignored in: <function Pool.__del__ at 0x7ff13bfd6b00>
E               
E               Traceback (most recent call last):
E                 File "/usr/local/lib/python3.10/multiprocessing/pool.py", line 265, in __del__
E                   _warn(f"unclosed running multiprocessing pool {self!r}",
E               ResourceWarning: unclosed running multiprocessing pool <multiprocessing.pool.ThreadPool state=RUN pool_size=4>
```



### Expected behavior

Not trigger the `Pool.__del__` warning.

### Installed versions

<details>

```
---Version info---
Polars: 0.14.22
Index type: UInt32
Platform: Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.31
Python: 3.10.5 (main, Jul  8 2022, 14:32:56) [GCC 10.2.1 20210110]
---Optional dependencies---
pyarrow: 9.0.0
pandas: 1.4.4
numpy: 1.23.3
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: 0.8
matplotlib: 3.6.0
```

</details>

--- a/py-polars/polars/internals/construction.py
+++ b/py-polars/polars/internals/construction.py
@@ -530,14 +530,13 @@ def dict_to_pydf(
         return PyDataFrame(data_series)
 
     if _NUMPY_AVAILABLE:
-        all_numpy = True
+        count_numpy = 0
         for val in data.values():
             # only start a thread pool from a reasonable size.
-            all_numpy = all_numpy and isinstance(val, np.ndarray) and len(val) > 1000
-            if not all_numpy:
-                break
+            count_numpy += int(isinstance(val, np.ndarray) and len(val) > 1000)
 
-        if all_numpy:
+        # if we have more than 3 numpy arrays we multi-thread
+        if count_numpy > 2:
             # yes, multi-threading was easier in python here
             # we cannot run multiple threads that run python code
             # and release the gil in pyo3
@@ -547,11 +546,11 @@ def dict_to_pydf(
             import multiprocessing.dummy
 
             pool_size = threadpool_size()
-            pool = multiprocessing.dummy.Pool(pool_size)
-            data_series = pool.map(
-                lambda t: pli.Series(t[0], t[1])._s,
-                [(k, v) for k, v in data.items()],
-            )
+            with multiprocessing.dummy.Pool(pool_size) as pool:
+                data_series = pool.map(
+                    lambda t: pli.Series(t[0], t[1])._s,
+                    [(k, v) for k, v in data.items()],
+                )
             return PyDataFrame(data_series)
 
     # fast path

